src/optim/FishLeg/fishleg.py

```diff
@@ -278,34 +283,99 @@ def update_aux(self) -> None:
                 u2 = u2 + torch.sum(params.grad * params.grad)
         u_norm = torch.sqrt(u2)
 
-        reg_term = 0.0
-        quad_term = 0.0
-        linear_term = 0.0
+        aux_loss = 0
 
-        p_idx = 0
-        for module in self.model.modules():
-            if not isinstance(module, nn.Sequential) and hasattr(module, "Qv"):
-                layer_grads = []
-                for name, param in module.named_parameters():
-                    if "fishleg_aux" not in name:
-                        layer_grads.append(param.grad.data / u_norm)
+        if self.method == "antithetic":
+            nat_grads = []
+            p_idx = 0
+            for module in self.model.modules():
+                if not isinstance(module, nn.Sequential) and hasattr(module, "Qv"):
+                    layer_grads = []
+                    for name, param in module.named_parameters():
+                        if "fishleg_aux" not in name:
```

Comment:
can we have flags for the parameters as an attribute? or do we need to have a new class definition? I am not a very big fan of replying on the name of the parameter to check if it is a auxiliary parameter or not. I upderstand it is not high priority at all to improve this, and we can just go on as it is. We use this condition so often that it might be even worth having a new function that filters this out in advance for polishness purposes. But again. Not priority. We can have a chat about this anyway and decide together what we think might be worth touching or not